### PR TITLE
Reduce the number of font-sizes used by AMO

### LINKED ISSUE #10250: Try to reduce the number of font-sizes used by AMO
This is a follow-up to https://github.com/mozilla/addons-frontend/issues/9078.

We are using a number of different font-sizes on AMO, over and above the defaults defined at https://github.com/mozilla/addons-frontend/blob/2021.03.18-1/src/amo/css/inc/vars.scss#L26-L33. 

There are a couple of parts to this:

1. We have some other font size variables defined in `vars.scss` (e.g., `$font-size-form-header`, `$font-size-puffy-buttons`, `$font-size-hero-m`, etc.).
2. There are explicit font-sizes in a number of components' scss files (e.g., `.Footer-links-header` has `font-size: 16px;` in `Footer/styles.scss`).

This issue is about looking at all of these instances and determining whether we can use one of the default font-sizes in place of these "other" font-sizes. In some cases we may be able to and in other cases me may not be able to.

### LINKED ISSUE #10264: Some layout issues should be fixed as a result of increased font size
Some layout issues should be fixed as a result of increasing the font-size, it was discussed in https://github.com/mozilla/addons-frontend/issues/9078#issuecomment-805075678

1. The Register or Log in - seems to stand a bit higher than EW and Dev Hub in the header - font-size at 11px

Before increasing the font sizes:

log out before

Now:
log out now

"Download Firefox" on Chrome is at 11 px too:

Download Firefox

2.  At the footer of the page there's a 18px "Add-ons" then "Browsers" and "Products" is at 16 px

footer

3. edit user profile page has some 10 px font which looks very small in contrast with the increased fonts

edit profile page

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10250
Fixes #10264 
~~Depends on #10467 (not really, but I built it on top of that)~~

The removes all of the specific font sizes from the stylesheets so we only use the pre-defined font sizes. 

@willdurand I didn't update `blog-utils/StaticAddonCard`, because I'm not sure where those numbers come from, but you may want to update it as a follow-up.

### CODE REVIEW

**Comment by willdurand** [comment]
> @willdurand I didn't update `blog-utils/StaticAddonCard`, because I'm not sure where those numbers come from, but you may want to update it as a follow-up.

Which ones? Do you meean the font-sizes for `.StaticAddonCard-error-overlay p` and `.AddonTitle`?

**Comment by bobsilverberg** [comment]
> > @willdurand I didn't update `blog-utils/StaticAddonCard`, because I'm not sure where those numbers come from, but you may want to update it as a follow-up.
> 
> Which ones? Do you meean the font-sizes for `.StaticAddonCard-error-overlay p` and `.AddonTitle`?

Yes, exactly those ones.

**Comment by willdurand** [comment]
> Yes, exactly those ones.

Ah ok, I think we are fine not updating those ones then. I'll double check, though.

File: src/blog-utils/StaticAddonCard/styles.scss [hunk 1]
```diff
@@ -79,7 +79,7 @@ $icon-size: 48px;
   & .AddonTitle-author,
   & .AddonTitle-author a,
   & .AddonTitle-author a:link {
-    font-size: $font-size-m;
+    font-size: $font-size-default;
     line-height: 1;
   }
```
Comment (willdurand) [inline src/blog-utils/StaticAddonCard/styles.scss:85]: At the bottom of the file, we're using `24px`. We could use `$font-size-heading-xs` instead.

I'll take care of the other one (`20px` in `.AddonTitle`).

**Review by willdurand — APPROVED** [review]
This is looking pretty good and I didn't find any (obvious) UI regressions. I see a few differences but they all look sane and better than before. Thanks!

r+wc

**Pull request merged**